### PR TITLE
Add convenience functions for testing

### PR DESCRIPTION
**Pull request by torepettersen**
Adding tests and fixing remarks from https://github.com/actix/actix-web/pull/1398

### CODE REVIEW

**Comment by JohnTitor** [comment]
Looks good! Could you also update changelog?

**Comment by torepettersen** [comment]
Yeah, I can.

**Comment by JohnTitor** [comment]
master should fix the build, could you rebase?

File: src/test.rs [hunk 1]
```diff
@@ -1068,6 +1143,28 @@ mod tests {
         assert_eq!(&result.id, "12345");
     }
 
+    #[actix_rt::test]
+    async fn test_body_json() {
```
Comment (pfrenssen) [inline src/test.rs:1147]: Since the function is called `read_body_json()` it is probably a good idea to call this test `test_read_body_json()`.

Comment (JohnTitor) [reply src/test.rs:1147]: It's the same as `read_response_json` and `test_response_json`, we can follow-up later.

**Review by JohnTitor — APPROVED** [review]
Sorry for the delay, thanks!

**Pull request merged**